### PR TITLE
added support for non-gregorian calendar to formatter

### LINKED ISSUE #8687: jalali date in formatter
if it's possible for  you add an ablility to framework, to have jalali date format and  when we set some property in config, like (dateType => 'jalali',) or (dateType => 'gregorian ')
by depending on this property in config, formatter works.

hear is the best jalalidate vendor in php that tested in all iranian app and it's working currectly:
http://jdf.scr.ir/

### PR DESCRIPTION
**Pull request by cebe**
fixes #8687 

Need to verify this actually works.

@hooman-pro @z-avanes could you provide information about calendars and expected formatting?

### CODE REVIEW

**Comment by cebe** [comment]
Seems there are a lot of different calendars out there. https://en.wikipedia.org/wiki/Template:Calendars would be good if we could cover the most relevant here.

**Comment by z-avanes** [comment]
yes a lot of calendars exist and i think ICU project cover all ( http://www.icu-project.org/ )
i use new PR code in my local machine and its ok and verfiy for persian calender ( probably its ok for other calender supported by intl )

File: tests/framework/i18n/FormatterDateTest.php [hunk 1]
```diff
@@ -88,6 +88,32 @@ public function testAsDate()
...
 
+    public function testIntlAsDateOtherCalendars()
+    {
+        // Persian calendar
+        $this->formatter->locale = 'fa_IR@calendar=persian';
+        $this->formatter->calendar = \IntlDateFormatter::TRADITIONAL;
+        $this->formatter->timeZone = 'UTC';
+
+        $value = 1451606400; // Fri, 01 Jan 2016 00:00:00 (UTC)
+        $this->assertSame('۱۳۹۴/۱۰/۱۱ ه‍.ش.', $this->formatter->asDate($value, 'short'));
```
Comment (cebe) [inline tests/framework/i18n/FormatterDateTest.php]: this fails on travis but works on my computer, seems icu version makes this untestable...

**Comment by cebe** [comment]
merged. @z-avanes thanks for checking!

**Comment by z-avanes** [comment]
thanks too @cebe

**Pull request closed without merging**